Tools/Replay/DataFlashFileReader.cpp

```diff
@@ -55,12 +83,15 @@ bool DataFlashFileReader::update(char type[5])
     uint8_t msg[f.length];
 
     memcpy(msg, hdr, 3);
-    if (::read(fd, &msg[3], f.length-3) != f.length-3) {
+    if (read_input(&msg[3], f.length-3) != f.length-3) {
         return false;
     }
 
     strncpy(type, f.name, 4);
     type[4] = 0;
 
-    return handle_msg(f,msg);
+    message_count++;
+    bool ret = handle_msg(f,msg);
+
+    return ret;
```

Comment:
why? just `return handle_msg()` was fine... it seems you only want to add the `message_count++` ?

Reply:
And I actually think the `message_count++` is tied to read_input() returning true. So it should be done just after that method is called rather than when this method is returned.  Or it could be moved inside read_input in case it succeeded.

Reply:
I agree with the `ret` not being needed, but I disagree that the `message_count` is tied to the `read_input()`: that method returns the read size, not a boolean, and just because it read something it doesn't mean a message is going to be successfully parsed.